llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -1413,50 +1413,97 @@ bool AMDGPUInstructionSelector::selectIntrinsicCmp(MachineInstr &I) const {
   return true;
 }
 
+// Ballot has to zero bits in input lane-mask that are zero in current exec,
+// Done as AND with exec. For inputs that are results of instruction that
+// implicitly use same exec, for example compares in same basic block, use copy.
+bool isBallotCopy(Register Reg, MachineRegisterInfo &MRI,
+                  MachineBasicBlock *MBB) {
+  MachineInstr *MI = MRI.getVRegDef(Reg);
+  // Look through copies, truncs and anyext. TODO: just copies
```

Comment:
What situations are there redundant copies that need to be looked through? I would assume any trivial copies to be optimized out

Reply:
non-trivial cross bank copies scc to vcc for example.

Reply:
That would be at most one copy? 

Also thinking about whether such a trivially uniform ballot should have folded out already